File: backend-go/internal/modules/system/entity/sys_login_log.go
package entity

import (
	"drip-admin/backend-go/internal/common"
	"time"
)

type SysLoginLog struct {
	ID            common.Int64String  `gorm:"column:id;primaryKey" json:"id"`
	UserID        *common.Int64String `gorm:"column:user_id" json:"userId"`
	Username      string              `gorm:"column:username" json:"username"`
	RealName      *string             `gorm:"column:real_name" json:"realName"`
	LoginType     string              `gorm:"column:login_type" json:"loginType"`
	Status        string              `gorm:"column:status" json:"status"`
	FailureReason *string             `gorm:"column:failure_reason" json:"failureReason"`
	IP            *string             `gorm:"column:ip" json:"ip"`
	UserAgent     *string             `gorm:"column:user_agent" json:"userAgent"`
	DeviceType    *string             `gorm:"column:device_type" json:"deviceType"`
	LoginAt       *time.Time          `gorm:"column:login_at" json:"loginAt"`
}

func (SysLoginLog) TableName() string { return "sys_login_log" }

File: backend-python/app/modules/system/entity/sys_login_log.py
from datetime import datetime

from sqlalchemy import BigInteger, DateTime, String
from sqlalchemy.orm import Mapped, mapped_column

from app.modules.system.entity.base import Base


class SysLoginLog(Base):
    __tablename__ = "sys_login_log"

    id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
    user_id: Mapped[int | None] = mapped_column(BigInteger, nullable=True)
    username: Mapped[str] = mapped_column(String(64))
    real_name: Mapped[str | None] = mapped_column(String(64), nullable=True)
    login_type: Mapped[str] = mapped_column(String(32))
    status: Mapped[str] = mapped_column(String(16))
    failure_reason: Mapped[str | None] = mapped_column(String(255), nullable=True)
    ip: Mapped[str | None] = mapped_column(String(64), nullable=True)
    user_agent: Mapped[str | None] = mapped_column(String(512), nullable=True)
    device_type: Mapped[str | None] = mapped_column(String(64), nullable=True)
    login_at: Mapped[datetime] = mapped_column(DateTime)
